include/swift/Driver/Compilation.h

```diff
@@ -330,9 +321,12 @@ class Compilation {
   const char *getAllSourcesPath() const;
 
   /// Asks the Compilation to perform the Jobs which it knows about.
+  ///
+  /// \param TaskQueue The TaskQueue used to schedule jobs for execution.
+  ///
   /// \returns result code for the Compilation's Jobs; 0 indicates success and
   /// -2 indicates that one of the Compilation's Jobs crashed during execution
-  int performJobs();
+  int performJobs(std::unique_ptr<sys::TaskQueue> &&TaskQueue);
```

Comment:
Two comments here:
1. Although it's fairly common practice in C++, I find it more readable to use a different identifier for the instance than for the type, perhaps `TheTaskQueue` or `TQ`?
2. Might a caller ever want to inspect the task queue after `performJobs` returns?

Reply:
1) Agreed

2) For my purposes and the purposes of the driver, both only need the `TaskQueue` to live as long as the Compilation session.  @graydon or @jrose-apple can probably think of concrete reasons why we may want to extend it in the future.

Reply:
I think to David's point there's no reason to have the Compilation *destroy* the TaskQueue, although it is going to irrevocably do a bunch of stuff to it. An lvalue reference (or a normal pointer) probably makes more sense.